### PR TITLE
WIP: Issue 374 zonal integration

### PR DESCRIPTION
**Pull request by DanielJDufour**
Hi, @wadhwamatic and @ericboucher .  This is a work in progress PR providing a demo of the new polygon intersection analysis.  It can help provide some points for discussion.

For the first test, I intersected all the tropical storm buffers with Myanmar districts.

Here's a screenshot of the table results
zonal-table

A few questions:
- Should area be converted to Square Kilometers?  What unit or units of measurement makes the most sense for your analysts?  In the short-term, we should probably choose just one unit of measurement.  In the long-term, we could discuss displaying a second column for another unit like hectares.
- How should we track the names of classes (like wind_speed)?  Hazard datasets can have a lot of different properties.  How should we track which class property to use for zonal analysis?  Is this something that should just live in the prism-frontend code or be in a config file?

Here's a screenshot of the results on the map:
zonal-map

A few questions:
- What color should we use to style the GeoJSON?  To start, I picked a range from 0 to 1 with zero being white and 1 being red.
- What property should we use for coloring?  We currently can use percentage (of zone intersecting hazard class) because it nicely fits with the 0-1 color range.  It certainly makes styling easier, but it also means that districts of largely different sizes may appear in the same color if they have the same percentage of overlap.  Alternatively, we could style by absolute area and rescale values to the true min/max range (but then hard hit but small districts might be overlooked).  Like the area units, we should probably choose one strategy to start, but could always discuss adding a second option in the future.
- If we want to hard code styling, where should that configuration live?

### CODE REVIEW

File: src/utils/admin-utils.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      return 'ST';
+  }
+}
+
+export function fetchAdminLayerGeoJSON(
+  adminLevel = 1,
+): Promise<FeatureCollection> {
+  const url = getAdminLayerURL(adminLevel);
+  return fetch(url).then(r => r.json());
+}
```
Comment (ericboucher) [inline src/utils/admin-utils.ts:32]: This doesn't scale/transfer to other countries, can you add more docstring as to what these functions are supposed to achieve?

**Comment by wadhwamatic** [comment]
Great to see a demo, thanks Daniel.

> * Should area be converted to Square Kilometers?  What unit or units of measurement makes the most sense for your analysts?  In the short-term, we should probably choose just one unit of measurement.  In the long-term, we could discuss displaying a second column for another unit like hectares.

I'd start with sq kms. Down road, hectares would make sense (and that's a reminder that adding croplands / crop masks as an input to the analysis workflow is something to keep in mind)

> * How should we track the names of classes (like wind_speed)?  Hazard datasets can have a lot of different properties.  How should we track which class property to use for zonal analysis?  Is this something that should just live in the prism-frontend code or be in a config file?

I don't think we have this condition yet or we've addressed it already. For the current 'exposure analysis' in Myanmar, we're using the wind buffer as an input, which has an attribute 'label' which provides the wind speed. In the config of the layer's exposure analysis, we could also specify another attribute. Can this be applied here as well?

BTW, I'm not understanding what the class:index column is describing. What do these values represent?
 
> * What color should we use to style the GeoJSON?  To start, I picked a range from 0 to 1 with zero being white and 1 being red.

Let's start with this style and see how it works for us: https://colorbrewer2.org/#type=sequential&scheme=Reds&n=5

> * What property should we use for coloring?  We currently can use percentage (of zone intersecting hazard class) because it nicely fits with the 0-1 color range.  It certainly makes styling easier, but it also means that districts of largely different sizes may appear in the same color if they have the same percentage of overlap.  Alternatively, we could style by absolute area and rescale values to the true min/max range (but then hard hit but small districts might be overlooked).  Like the area units, we should probably choose one strategy to start, but could always discuss adding a second option in the future.

Good point. I think starting with percentage makes sense but the caveat you mentioned is significant. Absolute area's caveats seem more of an issue in comparison though

> * If we want to hard code styling, where should that configuration live?

We also have a hard coded style for the population exposure results. It would be good if we could make that and these results a config in either prism.json, layers.json or some new file at the deployment/country level. It could include styles for various numbers of categories so we can add more classes as needed

File: src/components/Login/index.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {
   Button,
   Grid,
 } from '@material-ui/core';
-import { useMsal } from '@azure/msal-react';
+import { useMsal } from '@azure/msal-react'; // eslint-disable import/no-unresolved
```
[the remark this answers is not in the document]
Comment (DanielJDufour) [reply src/components/Login/index.tsx:10]: probably not.  good catch!

File: src/config/types.ts [hunk 3]
```diff
@@ -263,6 +264,24 @@ export class WMSLayerProps extends CommonLayerProps {
...
+  @makeRequired
+  title: string;
+
+  @optional
+  additionalQueryParams?: { [key: string]: string };
+
+  @optional
+  wcsConfig?: RawDataConfiguration;
+
+  @optional // If included, we infer the layer is a vector layer.
```
[the remark this answers is not in the document]
Comment (DanielJDufour) [reply src/config/types.ts:281]: yes that's correct, but honestly, I'm probably going to remove the WFSLayerProps from the this PR.  I originally thought I'd have to build out a lot of WFS stuff and configure that independent of the WMS config stuff and interfaces.  But I opted to dynamically create the WFS information from the WMS info to keep things simpler and have a single source of truth.  Happy to discuss more.

File: src/utils/admin-utils.ts (hunk 1 above)
Comment (DanielJDufour) [reply src/utils/admin-utils.ts:32]: you are correct. I hardcoded it to start, so we can get a visible prototype we can discuss.  Before making this PR ready for review, we'll have to decide where this configuration mapping of country to admin property and file name should live.

**Pull request closed without merging**